File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums: 
            return 0
        
        nums.sort()
        longest_streak, current_streak = 1, 1

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1] + 1:
                current_streak += 1
                longest_streak = max(longest_streak, current_streak)
            elif nums[i] == nums[i - 1]:
                continue
            else:
                current_streak = 1
        
        return longest_streak

File: Data Structures & Algorithms/house-robber/submission-2.cpp
class Solution {
private:
    int solve(const vector<int>& nums, vector<int>& memo, int i) {
        if (i >= nums.size()) return 0;

        if (memo[i] != -1) return memo[i];

        int rob = nums[i] + solve(nums, memo, i + 2);
        int not_rob = solve(nums, memo, i + 1);
        return memo[i] = max(rob, not_rob);
    }

public:
    int rob(vector<int>& nums) {
        vector<int> memo(nums.size(), -1);
        return solve(nums, memo, 0);
    }
};
